### PR TITLE
ADEN-4876 adjust tests to new VUAP

### PR DESCRIPTION
**Pull request by bulaszenko**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 1]
```diff
@@ -190,21 +213,25 @@ private WebElement findCloseButton() {
...
-  private void waitFor(final Predicate<AutoplayVuap> predicate, final long timeout) {
-    new FluentWait<>(this)
-        .withTimeout(timeout, TimeUnit.SECONDS)
-        .pollingEvery(1, TimeUnit.SECONDS)
-        .until(predicate);
-  }
-
-  private boolean isOverlayVisible() {
-    return driver.findElement(By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot))).isDisplayed();
+  private boolean isOverlayNoVisible() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: MAJOR Remove this unused private "isOverlayNoVisible" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java [hunk 2]
```diff
@@ -1,143 +1,52 @@
 package com.wikia.webdriver.testcases.adstests;
 
-import com.wikia.webdriver.common.WindowSize;
 import com.wikia.webdriver.common.contentpatterns.AdsContent;
 import com.wikia.webdriver.common.core.Assertion;
-import com.wikia.webdriver.common.core.annotations.NetworkTrafficDump;
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java [hunk 3]
```diff
@@ -1,143 +1,52 @@
...
 import com.wikia.webdriver.common.core.url.Page;
 import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
 import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
-import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VideoFanTakeover;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.AutoplayVuap;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapAssertions;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapAssertions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java [hunk 4]
```diff
@@ -1,143 +1,52 @@
...
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/20afe8148c8d8338a2aa6f0d3141711e19b04fd8/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/20afe8148c8d8338a2aa6f0d3141711e19b04fd8/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/20afe8148c8d8338a2aa6f0d3141711e19b04fd8/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/0889d6cf77a237907590f85f53b2dddc2c770a20/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/0889d6cf77a237907590f85f53b2dddc2c770a20/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/0889d6cf77a237907590f85f53b2dddc2c770a20/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/2e6002781f2231e8cb50ec3747d4a47ac6c8d202/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/2e6002781f2231e8cb50ec3747d4a47ac6c8d202/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/2e6002781f2231e8cb50ec3747d4a47ac6c8d202/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 5]
```diff
@@ -190,21 +213,25 @@ private WebElement findCloseButton() {
...
   }
 
   public boolean isResolvedStateDisplayed(double defaultVideoHeight, double resolvedVideoHeight) {
     return EXPECTED_PERCENTAGE_DIFFERENCE_IN_VIDEO_AD_HEIGHT == getStatesPercentageDifference(defaultVideoHeight, resolvedVideoHeight);
   }
 
+  public boolean isVideoAdBiggerThanImageAd(double videoHeight, double imageHeight) {
+    int percentResult = (int)Math.round(100-(100/(videoHeight/imageHeight)));
+    if (percentResult == PERCENTAGE_DIFFERENCE_BETWEEN_VIDEO_AND_IMAGE_AD) {
+      return true;
```
Comment (v3rth) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: you can use result of comparison as returned value

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/5ddf57211e5905a7889b61016859fb37dfd9e70e/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/5ddf57211e5905a7889b61016859fb37dfd9e70e/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/5ddf57211e5905a7889b61016859fb37dfd9e70e/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/38184571b23e2c30199e354cad85ec68a45c16b5/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/38184571b23e2c30199e354cad85ec68a45c16b5/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/38184571b23e2c30199e354cad85ec68a45c16b5/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 6]
```diff
@@ -3,6 +3,7 @@
 import com.google.common.base.Predicate;
 import com.wikia.webdriver.common.core.WikiaWebDriver;
 import com.wikia.webdriver.common.core.elemnt.Wait;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.logging.PageObjectLogging'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java [hunk 7]
```diff
@@ -196,21 +219,19 @@ private WebElement findCloseButton() {
...
+    return wait.forElementNotVisible(By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot)));
   }
 
   public boolean isResolvedStateDisplayed(double defaultVideoHeight, double resolvedVideoHeight) {
     return EXPECTED_PERCENTAGE_DIFFERENCE_IN_VIDEO_AD_HEIGHT == getStatesPercentageDifference(defaultVideoHeight, resolvedVideoHeight);
   }
 
+  public boolean isVideoAdBiggerThanImageAd(double videoHeight, double imageHeight) {
+    int percentResult = (int)Math.round(100-(100/(videoHeight/imageHeight)));
+    return (percentResult == PERCENTAGE_DIFFERENCE_BETWEEN_VIDEO_AND_IMAGE_AD);
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java]: MAJOR Remove those useless parentheses. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/2905a9e206a0f9df279123dbc480f6495f9305fb/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/2905a9e206a0f9df279123dbc480f6495f9305fb/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/2905a9e206a0f9df279123dbc480f6495f9305fb/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/fc1913f6179613bff73b7173474663e3c9814fa4/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/fc1913f6179613bff73b7173474663e3c9814fa4/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/fc1913f6179613bff73b7173474663e3c9814fa4/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AdsDataProvider.java#L1256](https://github.com/Wikia/selenium-tests/blob/1ac176958eee6bc633f7f3b846e89ba0d56bc6a8/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L1256): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L4](https://github.com/Wikia/selenium-tests/blob/1ac176958eee6bc633f7f3b846e89ba0d56bc6a8/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L4): Remove this unused import 'com.wikia.webdriver.common.core.elemnt.JavascriptActions'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [TestAdsVuapTheNewGeneration.java#L10](https://github.com/Wikia/selenium-tests/blob/1ac176958eee6bc633f7f3b846e89ba0d56bc6a8/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapTheNewGeneration.java#L10): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.componentobject.ad.VuapVideos'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**